skrf/network.py

```diff
@@ -1141,6 +1144,74 @@ def f(self, f):
         tmpUnit = self.frequency.unit
         self.frequency = Frequency.from_f(f, unit=tmpUnit)
 
+    @property
+    def f_noise(self):
+      """
+      the frequency vector for the noise of the network, in Hz.
+      """
+      return self.freq_noise
+
+    @property
+    def y_opt(self):
+      """
+      the optimum source admittance to minimize noise, relative to the frequency values in self.noise_freq
+      """
+      if self.noise is None:
+        raise ValueError('network does not have noise')
+      return (npy.sqrt(self.noise[:,1,1]/self.noise[:,0,0] - npy.square(npy.imag(self.noise[:,0,1]/self.noise[:,0,0])))
+          + 1.j*npy.imag(self.noise[:,0,1]/self.noise[:,0,0]))
+
+    @property
+    def z_opt(self):
+      """
+      the optimum source impedance to minimize noise, relative to the frequency values in self.noise_freq
+      """
+      return 1./self.y_opt
+
+    @property
+    def nfmin(self):
+      """
+      the minimum noise figure for the network, relative to the frequency values in self.noise_freq
+      """
+      if self.noise is None:
+        raise ValueError('network does not have noise')
+
+      k = 1.38064852e-23
+      T0 = 290.
+      return npy.real(1. + (self.noise[:,0,1] + self.noise[:,0,0] * npy.conj(self.y_opt))/(2*k*T0))
+
+    @property
+    def nfmin_db(self):
+      """
+      the minimum noise figure for the network in dB, relative to the frequency values in self.noise_freq
+      """
+      return 10.*npy.log10(self.nfmin)
+
+    def nf(self, z):
+      """
+      the noise figure for the network if the source impedance is z
+      """
+      # TODO interpolate/broadcast/whatever z0 to get the right vector here
+      z0 = self.z0
+      y_opt = self.y_opt
+      fmin = self.nfmin
+      rn = self.rn
+
+      ys = 1./z
+      gs = npy.real(ys)
+      return fmin + rn/gs * npy.square(npy.absolute(ys - y_opt))
+ 
+    @property
+    def rn(self):
+      """
+      the equivalent noise resistance for the network, relative to the frequency values in self.noise_freq
+      """
+      if self.noise is None:
+        raise ValueError('network does not have noise')
+      k = 1.38064852e-23
```

Comment:
would be good to define these constants once and import them to prevent errors,

Reply:
Done!